test/test_resource.rs

```diff
@@ -1,7 +1,22 @@
 use nix::sys::resource::{Resource, getrlimit, setrlimit};
 
 #[test]
-pub fn test_resource_limits() {
+pub fn test_resource_limits_nofile() {
+    let (soft_limit, hard_limit) = getrlimit(Resource::RLIMIT_NOFILE).unwrap();
+
+    // make sure the soft limit and hard limit are not equal
+    let soft_limit = match soft_limit {
+        Some(nofile) => Some(nofile -1),
+        None => Some(1024),
+    };
+    setrlimit(Resource::RLIMIT_NOFILE, soft_limit, hard_limit).unwrap();
+
+    let (new_soft_limit, new_hard_limit) = getrlimit(Resource::RLIMIT_NOFILE).unwrap();
+    assert!(new_soft_limit != new_hard_limit);
```

Comment:
This should be `assert!(new_soft_limit, soft_limit)`

Reply:
Done